plugin/src/main/groovy/com/novoda/staticanalysis/internal/lint/LintConfigurator.groovy

```diff
@@ -28,16 +30,30 @@ class LintConfigurator implements Configurator {
 
     @Override
     void execute() {
-        project.extensions.findByType(StaticAnalysisExtension).ext."lintOptions" = { Closure config ->
-            if (!isAndroidProject(project)) {
-                return
+        project.extensions.findByType(StaticAnalysisExtension).ext.lintOptions = { Closure config ->
+            project.plugins.withId('com.android.application') {
+                configureWithVariants(config, filteredApplicationVariants)
             }
-            configureLint(config)
-            configureToolTask()
+            project.plugins.withId('com.android.library') {
+                configureWithVariants(config, filteredLibraryVariants)
+            }
+
+        }
+    }
+
+    private void configureWithVariants(Closure config, DomainObjectSet variants) {
+        configureLint(config)
+        if (includeVariantsFilter != null) {
+            variants.all { configureCollectViolationsTask(it) }
+        } else {
+            configureCollectViolationsTask()
         }
     }
 
     private void configureLint(Closure config) {
+        project.android.lintOptions.ext.includeVariants = { Closure<Boolean> filter ->
```

Comment:
@mr-archano 
Could u explain us what exactly happens in this closure?

Reply:
I'm not sure what you're asking @tobiasheine (as I didn't write this), but I will try to explain what I can see going on here: L54 adds an extension method to `lintOptions` that takes a `filter` predicate, used by `VariantAware.variantSelector` to actually filter in only the variants for which `filter` is evaluated as `true`

Reply:
Right, he was asking because this is a copy paste from the other ones.
I was also wondering how this works exactly. One question comes to mind, there is no mention of the input of the closure. We expect the input to be a variant but I guess there is no limit? Can it be anything?

Reply:
I understand that we define a extension `includeVariants` with a `Closure<Boolean>` and what it is used for.
What is confusing to me is the assignment `includeVariantsFilter = filter` within the closure or when it is executed.  Is it like a callback that is triggered when the user specifies the extension?